### DISCUSSION TITLE
Zwave JS API o Zwave JS UI + MQTT?

### DISCUSSION THREAD

**Discussion by Whixter** [indomus/forum#2574] [ZigBee, Z-Wave, Thread, Matter]
Carissimi,
dopo ormai 3 anni di sperimentazioni ed innumerevoli integrazioni ed aggiornamenti su HA ed il mio Pi4, al mio ultimo aggiornamento del firmware del router fritzBox + contemporaneo aggiornamento di HA Core su Docker e rispettivi docker annessi (Mosquitto, Zigbee2MQTT, Zwave2MQTT, etc...) mi sono ritrovato una serie infinita di inspiegabili problemi di autenticazione ad HA da alcuni device quando connessi in WI-FI e dopo aver perso un paio di giorni a googlare e fare debug senza successo ho deciso che la cosa migliore fosse finalmente di ripartire da zero nell'installazione di tutto il sistema con relative integrazioni.

Al netto del fatto che solo un folle poteva lanciarsi nell'aggiornamento di tutte queste variabili in contemporanea senza attendersi problemi 😄 , in realtà era più di un anno che avrei voluto ripartire da zero nell'installazione del mio hub con l'esperienza di oggi e probabilmente questo è stato semplicemente un modo inconscio per trovarmi costretto ad affrontare il problema 😅 

Lato positivo: mi ritroverò, spero, un sistema finalmente più pulito e funzionante, non più "per tentativi", ma per presa coscienza del funzionamento di ciascuna integrazione e del codice.

Una delle cose che ho scoperto in questi 3 anni è che non esiste una sola opzione per fare le cose. Esiste la più facile ed intuitiva (certamente non lo era l'installazione di core su docker da neofita linux 😄) ma esiste anche la più complessa ma flessibile e performante ed è qui che nasce il mio dubbio e la mia domanda del post (Sorry per la lunga introduzione ma volevo dare il contesto):

Sulla nuova installazione ho già istanziato mosquitto (Broker MQTT) su docker perchè nelle mie precedenti sperimentazioni avevo utilizzato inizialmente senza grande soddisfazione l'integrazione Deconz ed infine invece passando a Zigbee2MQTT mi sono trovato alla grande.

Per ZWAVE ho iniziato invece 3 anni fa con OpenZwave migrando poi Zwave JS e Zwave2mqtt utilizzando però l'API.
Ora che sto reinstallando il dubbio amletico che ho è il seguente:

**E' meglio utilizzare l'API (Zwave JS) oppure è meglio usare MQTT (ZWave JS UI) anche per Zwave? 
Le due soluzioni sono sovrapponibili in termini di funzionalità (aggiornamenti OTA etc...) e prestazioni (sia in termini di carico sull'integrazione - e quindi HA - sia in termini di traffico rete - Chiamate API vs MQTT) oppure una è meglio dell'altra?**

Ho provato a cercare un po' in giro ed ho visto che è stata fatta un po' di confusione con il nome del bridge (che citava mqtt nel nome ma si poteva utilizzare anche solo l'api), varie diatribe sulla maggiore semplicità d'uso dell'API ma nessuna vera valutazione di performance e flessibilità.

**Qualcuno sa dirmi, magari per esperienza o ancor meglio sulla base di conoscenze più approfondite delle mie su MQTT vs chiamate websocket e reti come sia meglio procedere?**

Da un punto di vista di pulizia avrei detto MQTT per tutto, ma non mi voglio complicare la vita andando incontro magari a limitazioni di funzionalità/compatibilità se non c'è un effettivo beneficio.

Grazie in anticipo a chiunque voglia trovare il tempo di rispondere e soprattutto grazie infinite al team Indomus.
Con le vostro guide sono passato da totale niubbo di linux a mezzo niubbo e questo è un traguardo eccezionale 😄 

Whixter 

P.S. ho cercato senza successo sia su Indomus che su Google per una simile discussione.
Chiedo venia in anticipo se è stata già ampiamente approfondita e me la sono persa.

**Comment by KC-inDomus** [comment]
[API](https://indomus.it/guide/integrare-componenti-z-wave-su-home-assistant-via-z-wave-js-server-base-o-zwavejs2mqtt/)

**Reply by Whixter** [reply]
Grazie KC, 
Avevo letto la guida, ma non mi ero riuscito a fare un'idea. 
Questo messaggio dice tutto in sole 3 lettere 😄: lo sforzo di configurazione manuale che affronterei con MQTT (almeno con il mio livello medio-basso di competenza in merito) non vale i benefici.

Ricevuto! 
Ripropongo dunque la stessa scelta che avevo fatto in precedenza ma questa volta coscentemente.
Grazie ancora

**Reply by KC-inDomus** [reply]
esatto ;) figurati, di cosa
